netbox/client.go

```diff
@@ -95,7 +120,38 @@ func (c *Client) NewRequest(method string, endpoint string, options Valuer) (*ht
 	}
 	u.RawQuery = values.Encode()
 
-	return http.NewRequest(method, u.String(), nil)
+	return http.NewRequest(method, u.String(), body)
+}
+
+// NewJSONRequest creates a HTTP request using the input HTTP method, URL
```

Comment:
How about:

```
// NewJSONRequest creates a HTTP request using the input HTTP method, URL
// endpoint, a Valuer which creates URL parameters for the request, and
// an io.Reader as the body of the request.
//
// If a nil Valuer is specified, no query parameters will be sent with the
// request.
//
// The body parameter is marshaled to JSON and sent as a HTTP request body.
// Body must not be nil.
```

This fixes up the grammar a bit, and since we don't allow the caller to set HTTP request headers anyway, the fact that it sets the Content-Type header is an implementation detail.